src/Nowruz/pages/jobs/detail/applicants/useApplicantAction.tsx

```diff
@@ -20,17 +30,55 @@ export const useApplicantAction = (
   const [columnVisibility, setColumnVisibility] = useState({
     select: false,
   });
-
+  const [searchTerm, setSearchTerm] = useState('');
+  const [pages, setPages] = useState([
+    { tab: 'applicants', page: 1 },
+    { tab: 'offered', page: 1 },
+    { tab: 'rejected', page: 1 },
+  ]);
+  const [totalCounts, setTotalCounts] = useState([
+    { tab: 'applicants', count: applicants.total_count },
+    { tab: 'offered', count: applicants.total_count },
+    { tab: 'rejected', count: applicants.total_count },
+  ]);
   const currentSelectedId = useRef<string>();
 
   const navigate = useNavigate();
 
   useEffect(() => {
+    setSearchTerm('');
```

Comment:
why can't we handle it by clicking on each tab and needing `useEffect`?

Reply:
If I want to handle it by clicking on tab, I have to change the basic component, `<HorizontalTab>`, because handleClick on tabs are inside the component. I'd rather not to do it.

Reply:
we should handle it via props but it's ok for now later we should refactor these redundant `useEffect`. less `useEffect`, easier and faster debugging